### PR TITLE
add coprocessor function week_with_mode

### PR DESCRIPTION
**Pull request by AbnerZheng**
## What have you changed? (mandatory)
This PR add the builtin funtion week_with_mode to Tikv, which is in the list of https://github.com/tikv/tikv/issues/3275, following the document of https://pingcap.com/blog-cn/30mins-become-contributor-of-tikv/.

## What are the type of the changes? (mandatory)
- New feature (non-breaking change which adds functionality)

## How has this PR been tested? (mandatory)
By unit tests.

## Does this PR affect documentation (docs/docs-cn) update? (mandatory)
No

## Does this PR affect tidb-ansible update? (mandatory)
No

## Refer to a related PR or issue link (optional)
https://github.com/tikv/tikv/issues/3275

### CODE REVIEW

File: src/coprocessor/dag/expr/builtin_time.rs [hunk 1]
```diff
@@ -664,4 +677,31 @@ mod tests {
...
         );
     }
+
+    #[test]
+    fn test_week_mode() {
+        let cases = vec![
+            ("2000-01-01", 0, 0i64),
+            ("2008-02-20", 0, 7i64),
+            ("2008-02-20 00:00:00", 1, 8i64),
+            ("2008-12-31 00:00:00", 1, 53i64),
```
Comment (ice1000) [inline src/coprocessor/dag/expr/builtin_time.rs]: Can you add some non-edge tests? Like non-leap years?

**Comment by ice1000** [comment]
Please signoff all your commits.

File: src/coprocessor/dag/expr/builtin_time.rs (hunk 1 above)
Comment (AbnerZheng) [reply src/coprocessor/dag/expr/builtin_time.rs]: Sure,  I will add more test cases.

Comment (ice1000) [reply src/coprocessor/dag/expr/builtin_time.rs]: 👍

**Comment by AbnerZheng** [comment]
Aha, I have add the tests but find it has been merged already. Let me add another PR.

**Pull request merged**